aregion.cpp

```diff
@@ -407,9 +407,9 @@ void ARegion::RunDecayEvent(Object *o, ARegionList *pRegs)
 	pFactions = PresentFactions();
 	forlist (pFactions) {
 		Faction *f = ((FactionPtr *) elem)->ptr;
-		f->Event(GetDecayFlavor() + *o->name + " " +
-				ObjectDefs[o->type].name + " in " +
-				ShortPrint(pRegs));
+		stringstream tmp;
+		tmp << GetDecayFlavor() << o->name << " " << ObjectDefs[o->type].name << " in " << ShortPrint(pRegs);
+		f->event(tmp.str());
```

Comment:
As I start moving things from Astring to string, these will crop up.. they will go away again as more and more of those fields move to string from Astring since strings have good support for appending things to them but char * (which half of these are) and AStrings are half-assed at best.  So any place you see me add something like this, expect it will eventually go away.